### PR TITLE
Documentation overhaul

### LINKED ISSUE #318: Serious lack of quality in documentation
Hello,

I was recently referred to this project by the team at Tokio. It looks very promising, however, the documentation suffers from a serious lack of quality and substance. 

There is no "hello world" example and basic steps to get an example working. The tutorial is beyond brief and contains many broken links. 

We've attempted to run the example projects by cloning the repo and ensuring all the dependencies are installed as per: https://fzyzcjy.github.io/flutter_rust_bridge/quickstart.html#-show-me-the-code

When executing the project using the command: _flutter_rust_bridge_codegen --rust-input frb_example/pure_dart/rust/src/api.rs --dart-output frb_example/pure_dart/lib/bridge_generated.dart --c-output frb_example/pure_dart/ios/Runner/bridge_generated.h_

the process panics with:

`[SEVERE] : Couldn't find dynamic library in default locations.
 [SEVERE] : Please supply one or more path/to/llvm in ffigen's config under the key 'llvm-path'.`

Please provide a simple step-by-step guide to setting up a **hello world** application. Please do not simply write refer to this and refer to that, and then see this over here and download this here and hope it works. This is an incredibly unprofessional way to present your tool and makes it very difficult to take your project seriously as a professional tool.

### PR DESCRIPTION
**Pull request by Desdaemon**
Resolves #318.

Address potential problems more plainly without having to depend on a troubleshooting section, and make it easier to start working without having to do too much infra setup. Some items on the list:
- [x] Create a project template at [flutter_rust_bridge_template](https://github.com/Desdaemon/flutter_rust_bridge_template)
- [x] Ensuring `flutter run` works for all platforms
- [x] Document steps needed to `flutter run` on all platforms
- [x] Write a hands-on example how to generate new code
- [x] Write a guide on how to integrate with existing Flutter projects
- [x] ~~Address possible issues during the initial iOS build~~
- [x] Create a short syntax guide of api.rs

---

Some tiny things added by @fzyzcjy (added here since only the root comment's checklist appears in the PR title as a tracker)

- [x] Explain a bit about #319
- [x] Remind users of running `build_runner` #330
- [x] Maybe this comment https://github.com/fzyzcjy/flutter_rust_bridge/issues/318#issuecomment-1037898017

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Excellent job! As for the last part, there are already some guides: https://fzyzcjy.github.io/flutter_rust_bridge/set_up_from_scratch.html So maybe we can make use of it

**Comment by ramanverma2k** [comment]
Looks great, @Desdaemon did an outstanding job.

**Comment by fzyzcjy** [comment]
Since this PR is ongoing, I do not edit the book at master branch but add an item in the checklist of this PR:

> Explain a bit about https://github.com/fzyzcjy/flutter_rust_bridge/pull/319

**Comment by fzyzcjy** [comment]
Add a tiny line again:

> - Remind users of running `build_runner` #330

**Comment by fzyzcjy** [comment]
Btw should we mention https://github.com/fzyzcjy/flutter_rust_bridge/issues/108 in the troubleshooting section, or possibly the new tutorial section? Then we may close #108

**Comment by fzyzcjy** [comment]
Added: [explain a bit about ios + cocoapods](https://github.com/Desdaemon/flutter_rust_bridge/commit/b4c68f9fad0b41ef5a4e1e56d0817394a04935f4)

**Comment by fzyzcjy** [comment]
@Desdaemon 

This file https://github.com/fzyzcjy/flutter_rust_bridge/blob/8cd8efa2563b278151a0f609cb1ed9ae0cb09cfc/book/src/generate/macos_intel.png is 620KB in size. Then, when users clone the repo (e.g. want to do the tutorial), they will spend extra time and bandwidth for it; and this increases the repo size permanently by this much.

Thus, maybe we can make this file a bit smaller? Possibly using (1) pngquant (2) smaller image resolution

**Comment by Desdaemon** [comment]
> Added: [explain a bit about ios + cocoapods](https://github.com/Desdaemon/flutter_rust_bridge/commit/b4c68f9fad0b41ef5a4e1e56d0817394a04935f4)

I found it a bit strange that when I created the template using 2.10, there was no longer any mention of CocoaPods in the build config. Can someone knowledgeable about this tell me what changed in the last Flutter update? @fzyzcjy

**Comment by fzyzcjy** [comment]
> I found it a bit strange that when I created the template using 2.10, there was no longer any mention of CocoaPods in the build config. Can someone knowledgeable about this tell me what changed in the last Flutter update? 

Interesting... I am on 2.8 yet (since my app is in production and 2.10 does not contain big enough interesting features). 

https://github.com/flutter/flutter/blob/master/examples/api/ios/Podfile even on master branch I do see some podfile.

maybe create an issue in flutter's github asking this? and possibly ask how to migrate old apps to the new one (since cocoapod is really useful)

File: book/src/integrate/deps.md [hunk 1]
```diff
@@ -28,4 +28,18 @@ following dependencies are also needed:
 
 Their usage is explained in [Using `build_runner`](../generate/build_runner.md).
 
+## Rust dependencies
+
+Similar to Dart, Rust requires a few dependencies to get going:
+- `flutter_rust_bridge`
+- `anyhow`, which supplies a common `Result` type for all functions
```
[the remark this answers is not in the document]
Comment (Desdaemon) [reply book/src/integrate/deps.md:35]: I agree, makes it easier for us too.

**Comment by fzyzcjy** [comment]
> - [ ] Create a short syntax guide of api.rs

Btw I will do this tomorrow

**Comment by fzyzcjy** [comment]
>  Create a short syntax guide of api.rs

I have implemented the `feature/*` chapter.

**Comment by fzyzcjy** [comment]
Btw I am considering creating a post in Reddit r/rust and r/flutter after this PR is merged, since now we have much more features and much better documentations - people who originally came and left b/c lack of doc or lack of features may be able to use it now. What do you think about it?

**Comment by Desdaemon** [comment]
Yeah, I think it's a good idea--wouldn't hurt to have some serious feedback now.

**Comment by fzyzcjy** [comment]
I have created chapter "using template to create your own project" and merge three chapters into it, trying to make the doc more easier to read.

Now the wonderful user-guide part looks like:

image

**Comment by fzyzcjy** [comment]
Sounds great and looking forward to merging this PR!

**Comment by fzyzcjy** [comment]
🎉

**Comment by fzyzcjy** [comment]
I may send a post one or two days later, since today's https://www.reddit.com/r/rust/ already has tons of "some package has new release"-style posts

**Pull request merged**